.github/workflows/non_maintainer_check.yaml

```diff
@@ -23,11 +23,22 @@ jobs:
       env:
         GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         AUTHOR: ${{ github.event.pull_request.user.login }}
+        REPO_NAME: ${{ github.event.repository.full_name }}
         PR_NUMBER: ${{ github.event.pull_request.number }}
         TEMP_DIR: ${{ runner.temp }}
       run: |
-        cat <<EOF > "${TEMP_DIR}/greeting.md"
-        @${AUTHOR}
+        # sometimes the author association of event payload is not correct to handle it
+        # so we fetch the latest author association from the API
+        AUTHOR_ASSOCIATION=$(gh api "/repos/${REPO_NAME}/pulls/${PR_NUMBER}" --jq '.author_association')
+        echo "AUTHOR_ASSOCIATION: ${AUTHOR_ASSOCIATION}"
+        [ "${AUTHOR_ASSOCIATION}" = "COLLABORATOR" ] && exit 0
+        [ "${AUTHOR_ASSOCIATION}" = "MANNEQUIN" ] && exit 0
+        [ "${AUTHOR_ASSOCIATION}" = "MEMBER" ] && exit 0
+        [ "${AUTHOR_ASSOCIATION}" = "OWNER" ] && exit 0
```

Comment:
I found this job ran on this PR. It's my PR, and I'm a MEMBER, so it's not expected to run.
So I added these double-check lines.
https://github.com/pipe-cd/pipecd/actions/runs/15107247811